src/omv/img/jpeg.c

```diff
@@ -70,28 +70,53 @@ static uint8_t *get_mcu()
                 }
             }
             break;
-        case 1:
-            for (int y=jpeg_enc.y_offset; y<(jpeg_enc.y_offset + MCU_H); y++) {
-                for (int x=jpeg_enc.x_offset; x<(jpeg_enc.x_offset + MCU_W); x++) {
-                    if (x >= jpeg_enc.img_w || y >= jpeg_enc.img_h) {
-                        *Y0++ = 0;
-                    } else {
+        case 1: {
+            int dx=MCU_W, dy=MCU_H;
+            uint32_t *s32, *d32;
+            int bSmall = 0; // truncated MCU
+            if (jpeg_enc.x_offset+dx > jpeg_enc.img_w)
+                dx = jpeg_enc.img_w - jpeg_enc.x_offset;
+            if (jpeg_enc.y_offset+dy > jpeg_enc.img_h)
+                dy = jpeg_enc.img_h - jpeg_enc.y_offset;
+            if (dx != MCU_W || dy != MCU_H) { // partial MCU, fill with 0's to start
```

Comment:
Can you remove the bsmall variable and just use the if statement for two loops here. Would be easier to read.

Reply:
Fixed